starknet_py/hash/sierra_class_hash.py

```diff
@@ -0,0 +1,55 @@
+from typing import List
+
+from starkware.cairo.common.poseidon_hash import poseidon_hash_many
+
+from starknet_py.common import int_from_bytes
+from starknet_py.hash.utils import _starknet_keccak
+from starknet_py.net.client_models import NewContractClass, NewEntryPoint
+
+
+def compute_sierra_class_hash(sierra_contract_class: NewContractClass) -> int:
```

Comment:
Eventually this could be changed to accept `NewCompiledContract` (currently only on feat/declare-v2 branch) which has non-optional abi.

Reply:
About that... #835 exists

Reply:
Let's address #835 in the separate PR, but I definitely see the point in the renaming from `New...` now.